src/diffusers/models/attention_processor.py

```diff
@@ -269,10 +276,16 @@ def __init__(
         # We use the AttnProcessor2_0 by default when torch 2.x is used which uses
         # torch.nn.functional.scaled_dot_product_attention for native Flash/memory_efficient_attention
         # but only if it has the default `scale` argument. TODO remove scale_qk check when we move to torch 2.1
+        # If torch_xla is available, we use pallas flash attention kernel to improve the performance.
         if processor is None:
-            processor = (
-                AttnProcessor2_0() if hasattr(F, "scaled_dot_product_attention") and self.scale_qk else AttnProcessor()
-            )
+            if hasattr(F, "scaled_dot_product_attention") and self.scale_qk:
+                if is_torch_xla_available:
```

Comment:
Same here too. Does this need to be guarded with a version check too?

Reply:
Added the version check for torch_xla here too.